src/mpid/ch4/netmod/ofi/ofi_spawn.c

```diff
@@ -39,8 +42,21 @@ int MPIDI_OFI_dynamic_send(uint64_t remote_gpid, int tag, const void *buf, int s
     } while (!req.done && (timeout == 0 || time_gap < (double) timeout));
 
     if (!req.done) {
-        /* FIXME: better error message */
-        mpi_errno = MPI_ERR_PORT;
+        /* time out, let's cancel the request */
+        int rc;
+        rc = fi_cancel((fid_t) MPIDI_OFI_global.ctx[ctx_idx].tx, (void *) &req.context);
+        if (rc && rc != -FI_ENOENT) {
+            MPIR_ERR_CHKANDJUMP4(rc < 0, mpi_errno, MPI_ERR_OTHER, "**ofid_cancel",
+                                 "**ofid_cancel %s %d %s %s", __SHORT_FILE__, __LINE__, __func__,
+                                 fi_strerror(-rc));
+
+        }
+        while (!req.done) {
+            mpi_errno = MPIDI_OFI_progress_uninlined(vni);
+            MPIR_ERR_CHECK(mpi_errno);
+        }
+
+        mpi_errno = MPIX_ERR_TIMEOUT;
```

Comment:
Should this be a SETANDSTMT instead of returning the bare class? Otherwise I assume this will get reported as `MPI_ERR_OTHER` by an `MPIR_ERR_CHECK` macro after returning.

Reply:
There is exemption that `MPI_ERR_OTHER` will not overwrite previous error class: https://github.com/pmodels/mpich/blob/46aecd9032e3d3fcc9eee31270056e422709488c/src/mpi/errhan/errutil.c#L889-L900

Reply:
Is this issue resolved?